gapis/gfxapi/vulkan/draw_call_mesh.go

```diff
@@ -0,0 +1,412 @@
+	backingMem := boundIndexBuffer.BoundBuffer.Buffer.Memory
+	if backingMem == nil {
+		return []uint32{}
+	}
+	bufferMemoryOffset := uint64(boundIndexBuffer.BoundBuffer.Buffer.MemoryOffset)
+	indexBindingOffset := uint64(boundIndexBuffer.BoundBuffer.Offset)
+	// TODO(qining): Get the maximum size of the bound buffer here from BoundBuffer.Range.
+	offset := bufferMemoryOffset + indexBindingOffset
+
+	extractIndices := func(sizeOfIndex uint64) []uint32 {
+		indices := []uint32{}
+		start := offset + uint64(firstIndex)*sizeOfIndex
+		size := uint64(indexCount) * sizeOfIndex
+		end := start + size
+		indicesSlice := backingMem.Data.Slice(start, end, s)
+		for i := uint64(0); (i < size) && (i+sizeOfIndex-1 < size); i += sizeOfIndex {
+			index := int32(0)
+			for j := uint64(0); j < sizeOfIndex; j++ {
+				oneByte := int32(indicesSlice.Index(i+j, s).Read(ctx, nil, s, nil))
+				index += oneByte << (8 * j)
+			}
+			index += vertexOffset
+			if index < 0 {
+				// TODO(qining): The index value is invalid, need to emit error mesage
+				// here.
+				index = 0
+			}
+			indices = append(indices, uint32(index))
+		}
+		return indices
+	}
+
+	switch boundIndexBuffer.Type {
+	case VkIndexType_VK_INDEX_TYPE_UINT16:
+		return extractIndices(2)
+	case VkIndexType_VK_INDEX_TYPE_UINT32:
+		return extractIndices(4)
+	}
+	return []uint32{}
+}
+
+func getVertexBuffers(ctx context.Context, s *gfxapi.State,
+	vertexCount, firstVertex uint32) (*vertex.Buffer, error) {
+
+	if vertexCount == 0 {
+		return nil, fmt.Errorf("Number of vertices must be greater than 0.")
+	}
+
+	vb := &vertex.Buffer{}
+	// Get the current bound vertex buffers
+	lastDrawInfo := getStateObject(s).LastDrawInfo
+	attributes := lastDrawInfo.GraphicsPipeline.VertexInputState.AttributeDescriptions
+	bindings := lastDrawInfo.GraphicsPipeline.VertexInputState.BindingDescriptions
+	// For each attribute, get the vertex buffer data
+	for _, attributeIndex := range attributes.KeysSorted() {
+		attribute := attributes.Get(attributeIndex)
+		if !bindings.Contains(attribute.Binding) {
+			// TODO(qining): This is an error, should emit error message here.
+			continue
+		}
+		binding := bindings.Get(attribute.Binding)
+		if !lastDrawInfo.BoundVertexBuffers.Contains(binding.Binding) {
+			// TODO(qining): This is an error, should emit error message here.
+			continue
+		}
+		boundVertexBuffer := lastDrawInfo.BoundVertexBuffers.Get(binding.Binding)
+		vertexData, err := getVerticesData(ctx, s, boundVertexBuffer,
+			vertexCount, firstVertex, binding, attribute)
+		if err != nil {
+			return nil, err
+		}
+		translatedFormat, err := translateVertexFormat(attribute.Format)
+		if err != nil {
+			// TODO(qining): This is an error, should emit error message here
+			continue
+		}
+		name := fmt.Sprintf("binding:%v,location:%v", binding.Binding, attribute.Location)
```

Comment:
In the shader it will look something like
`layout (binding=0, location=0)` however like C code this is entirely up to the programmer.
```glsl
layout(binding = 0, location=1)
layout ( binding=0, location   =   1)
layout ( location = 0, binding = 1 ) 
```
whatever the programmer prefers, so we should probably just pick a convention that would work, personally I prefer
```glsl
layout(binding=0, location=0) 
```